lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryOutputWalker.php

```diff
@@ -85,13 +110,27 @@ public function __construct($query, $parserResult, array $queryComponents)
      * Walks down a SelectStatement AST node, wrapping it in a SELECT DISTINCT.
      *
      * @param SelectStatement $AST
+     * @param bool $addMissingItemsFromOrderByToSelect
      *
      * @return string
      *
      * @throws \RuntimeException
      */
-    public function walkSelectStatement(SelectStatement $AST)
+    public function walkSelectStatement(SelectStatement $AST, $addMissingItemsFromOrderByToSelect = true)
```

Comment:
@stof would it make sense to move the body of walkSelectStatement to a separate method, say doWalkSelectStatement(SelectStatement $AST, $addMissingItemsFromOrderByToSelect) and then have the main walkSelectStatement method just call that? Not sure that obfuscation would help anything but it does resolve the function signature weirdness.